lib/rules/imports-requiring-flow.js

```diff
@@ -0,0 +1,91 @@
+const path = require("path");
+
+const checkImport = (context, rootDir, importPath, node) => {
+    const modules = context.options[0].modules || [];
+
+    for (const mod of modules) {
+        if (importPath.startsWith(".")) {
+            const filename = context.getFilename();
+            const absImportPath = path.join(path.dirname(filename), importPath);
+            const absModPath = path.join(rootDir, mod);
+            if (absModPath === absImportPath) {
+                const message = `Importing "${importPath}" requires using flow.`;
+                context.report({
+                    node: node,
+                    message: message,
+                });
+                break;
+            }
+        } else {
+            if (mod === importPath) {
+                const message = `Importing "${importPath}" requires using flow.`;
+                context.report({
+                    node: node,
+                    message: message,
+                });
+                break;
+            }
+        }
+    }
+};
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "Require flow when using certain imports",
+            category: "flow",
+            recommended: false,
+        },
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    modules: {
+                        type: "array",
+                        items: {
+                            type: "string",
+                        },
+                    },
+                    rootDir: {
+                        type: "string",
+                    },
+                },
+            },
+        ],
+    },
+
+    create(context) {
+        const modules = context.options[0].modules || [];
+        const rootDir = context.options[0].rootDir;
+        if (!rootDir) {
+            throw new Error("rootDir must be set");
+        }
+        const source = context.getSource();
+        const filename = context.getFilename();
+
+        let usingFlow = false;
+
+        return {
+            Program(node) {
+                usingFlow = node.comments.some(
+                    comment => comment.value.trim() === "@flow",
+                );
+            },
+            ImportDeclaration(node) {
```

Comment:
So if I'm reading this correctly, we're not handling dynamic import calls, right? It feels like we'll want to handle that, as well!